### PR TITLE
qscreenshot: 1.0 -> unstable-2021-10-18

### PR DESCRIPTION
**Pull request by matthiasbeyer**
qt4 -> qt5

(cherry picked from commit b0823dbd819aa6cbbccf256f9a8754b536475063)
Signed-off-by: Matthias Beyer <mail@beyermatthias.de>

---

Extracted from #174634
because this can be PRed independently and can land earlier than the PR above.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [ ] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Review by bjornfor — COMMENTED** [review]
IMHO, moving a package from a released version to an unreleased one requires justification in the commit message. I'm generally against it but won't block merging.

**Comment by matthiasbeyer** [comment]
Well I agree partly... this is for an update to qt5 (see where it is extracted from). I would like to see a proper release, but I would like to see qt4 removed much better.

**Comment by bjornfor** [comment]
> (cherry picked from commit https://github.com/NixOS/nixpkgs/commit/b0823dbd819aa6cbbccf256f9a8754b536475063)

I don't think it makes sense to cherry-pick commits that are not in one of nixpkgs permanent branches (master or release-X.Y). To me it looks like dangling pointers.

File: pkgs/applications/graphics/qscreenshot/default.nix [hunk 1]
```diff
@@ -1,23 +1,34 @@
...
-  version = "1.0";
+  version = "unstable-2021-10-18";
 
-  src = fetchurl {
-    url = "mirror://sourceforge/qscreenshot/qscreenshot-${version}-src.tar.gz";
-    sha256 = "1spj5fg2l8p5bk81xsv6hqn1kcrdiy54w19jsfb7g5i94vcb1pcx";
+  src = fetchgit {
+    url = "https://git.code.sf.net/p/qscreenshot/code";
+    rev = "e340f06ae2f1a92a353eaa68e103d1c840adc12d";
+    sha256 = "0mdiwn74vngiyazr3lq72f3jnv5zw8wyd2dw6rik6dbrvfs69jig";
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/qscreenshot/default.nix:18]:
```suggestion
    hash = "sha256-L8pktNt5NTNjNryJ5jniv2wrhxMH05G/8vHZTY7lsVU=";
```

File: pkgs/applications/graphics/qscreenshot/default.nix [hunk 2]
```diff
@@ -1,23 +1,34 @@
...
 
-  buildInputs = [ dos2unix which qt ]
-    ++ lib.optional stdenv.isDarwin Carbon;
-
-  # Remove carriage returns that cause /bin/sh to abort
-  preConfigure = ''
-    dos2unix configure
-    sed -i "s|lrelease-qt4|lrelease|" src/src.pro
-  '';
+  preConfigure = "cd qScreenshot";
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/qscreenshot/default.nix:21]:
```suggestion
  sourceRoot = "${src.name}/qScreenshot/";
```

File: pkgs/applications/graphics/qscreenshot/default.nix [hunk 3]
```diff
@@ -1,23 +1,34 @@
...
 
+  nativeBuildInputs = [
+    cmake
+    qttools
+    wrapQtAppsHook
+  ];
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/qscreenshot/default.nix:27]:
```suggestion
  ];

```

File: pkgs/applications/graphics/qscreenshot/default.nix [hunk 4]
```diff
@@ -1,23 +1,34 @@
...
+  buildInputs = [
+    qtbase
+    qtx11extras
+  ];
```
Comment (AndersonTorres) [inline pkgs/applications/graphics/qscreenshot/default.nix:31]:
```suggestion
  ];

```

File: pkgs/top-level/all-packages.nix [hunk 5]
```diff
@@ -34651,10 +34651,7 @@ with pkgs;
 
   qsampler = libsForQt5.callPackage ../applications/audio/qsampler { };
 
-  qscreenshot = callPackage ../applications/graphics/qscreenshot {
-    inherit (darwin.apple_sdk.frameworks) Carbon;
```
Comment (AndersonTorres) [inline pkgs/top-level/all-packages.nix:34655]: No more Darwin quirks?

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/top-level/all-packages.nix (hunk 5 above)
Comment (matthiasbeyer) [reply pkgs/top-level/all-packages.nix:34655]: Forwarding this question to the author of the patch @ajs124

**Comment by matthiasbeyer** [comment]
@AndersonTorres thanks for your suggestions, will squash the updates before merge (but after the outstanding discussion)

**Review by AndersonTorres — APPROVED** [review]

**Review by AndersonTorres — APPROVED** [review]

**Comment by AndersonTorres** [comment]
Why is it failing on Darwin?

**Pull request closed without merging**